firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java

```diff
@@ -106,32 +119,43 @@ public void listenForNotifications() {
     try {
       scheduledExecutorService.awaitTermination(3L, TimeUnit.SECONDS);
     } catch (InterruptedException ex) {
-      Log.i(TAG, "Thread Interrupted.");
+      Log.d(TAG, "Thread Interrupted.");
     }
   }
 
   // Auto-fetch new config and execute callbacks on each new message
   private void handleNotifications(InputStream inputStream) throws IOException {
     BufferedReader reader = new BufferedReader((new InputStreamReader(inputStream, "utf-8")));
     String message;
+    String fullMessage = "";
     while ((message = reader.readLine()) != null) {
-      if (!message.contains("latestTemplateVersionNumber")) {
-        continue;
-      }
+      fullMessage += message;
+
+      if (message.contains("}")) {
+        fullMessage = parseMessage(fullMessage);
+        if (!fullMessage.isEmpty()) {
+          long targetTemplateVersion = configFetchHandler.getTemplateVersionNumber();
+          try {
+            JSONObject jsonObject = new JSONObject(fullMessage);
+            if (jsonObject.has("latestTemplateVersionNumber")) {
```

Comment:
Would make these strings constants in the class

Reply:
Updated.